### PR TITLE
Process and unittest

### PR DESCRIPTION
**Pull request by 0x050f**

### CODE REVIEW

**Comment by 0x050f** [comment]
TODO:
- [ ] https://github.com/Harthann/kfs/issues/8

File: srcs/proc/signal.rs [hunk 1]
```diff
@@ -1,7 +1,15 @@
 use crate::proc::Id;
 use crate::proc::process::{Process, MASTER_PROCESS};
+use crate::errno::ErrNo;
 
-#[derive(Copy, Clone, PartialEq)]
+pub type SigHandlerFn = extern "C" fn (i32);
```
Comment (Harthann) [inline srcs/proc/signal.rs]: You can use rust fn instead, saving some unsafe call and allowing use of closure

File: srcs/kinit.rs [hunk 2]
```diff
@@ -160,19 +156,148 @@ pub extern "C" fn kinit() {
...
+		);
+	} else {
+		kprintln!(
+			"exited process pid: {} - signal: {}",
+			test,
+			__WSTOPSIG!(wstatus)
+		);
+	}
+	// 	loop {}
+}
```
Comment (Harthann) [inline srcs/kinit.rs]: I prefer to keep kinit clean and have only functions that initialise the kernel, dumb_mains should be declare inside kmain.

File: srcs/proc/mod.rs [hunk 3]
```diff
@@ -36,10 +40,10 @@ pub unsafe extern "C" fn wrapper_fn() {
 		"
 	mov eax, [esp]
```
Comment (Harthann) [inline srcs/proc/mod.rs]: You can add function parameter to make clear that this funciton waits a parameter.

**Review by Harthann — APPROVED** [review]

**Pull request merged**